src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -764,7 +765,16 @@ class LicenceService(
     standardConditionRepository.saveAll(standardConditions)
     bespokeConditionRepository.saveAll(bespokeConditions)
 
-    val additionalConditions = licence.additionalConditions.map {
+    val licenceConditions: List<AdditionalCondition> =
+      if (newStatus == VARIATION_IN_PROGRESS) {
+        licence.additionalConditions.filter {
+          !(LicenceType.valueOf(it.conditionType!!) == LicenceType.AP && licence.typeCode == LicenceType.AP_PSS && licence.isInPssPeriod())
```

Comment:
this would be easier to reason about if we moved the ` licence.typeCode == LicenceType.AP_PSS && licence.isInPssPeriod()` into the surrounding if block? 

those are specific to the licence rather than each condition